lib/SILGen/SILGenExpr.cpp

```diff
@@ -3015,9 +3015,37 @@ visitValueAndGradientExpr(ValueAndGradientExpr *E, SGFContext C) {
   return emitGradientInst(*this, C, E, /*isValueAndGrad*/ true);
 }
 
+// SWIFT_ENABLE_TENSORFLOW
 RValue RValueEmitter::
 visitAdjointExpr(AdjointExpr *E, SGFContext C) {
-  llvm_unreachable("handle #adjoint");
+  ConcreteDeclRef adjointFunc = E->getAdjointFunction();
+  FuncDecl *adjointDecl = cast<FuncDecl>(adjointFunc.getDecl());
+  SILLocation loc(adjointDecl);
+  SILDeclRef adjointDeclRef(adjointDecl);
+
+  // If adjoint has multiple parameter lists, mark as curried.
+  if (adjointDecl->getNumParameterLists() == 2) {
```

Comment:
>= 2?

this will also better match the comment above

Reply:
Done. Existing code uses `getNumParameterLists() == 2`, however.

Reply:
I mean ">= 2" -- the formatter ate ">" in my previous message :)